### PR TITLE
add raptor

### PR DESCRIPTION
**Pull request by logan-markewich**
Adds a raptor pack, implementing the RAPTOR paper

Credits to
- Original PR: https://github.com/run-llama/llama-hub/pull/924
- Original Authors/Code: https://github.com/parthsarthi03/raptor

### CODE REVIEW

File: llama-index-packs/llama-index-packs-raptor/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Raptor Retriever LlamaPack
+
+## Embedded Tables Retriever Pack w/ Unstructured.io
```
Comment (jerryjliu) [inline llama-index-packs/llama-index-packs-raptor/README.md:3]: nit: title is wrong

File: llama-index-packs/llama-index-packs-raptor/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+The `RaptorPack` comes with the `RaptorRetriever`, which offers ways of saving/reloading!
+
+If you are using a remote vector-db, just pass it in
+
+```python
+# Pack usage
+pack = RaptorPack(..., vector_store=vector_store)
+
+# RaptorRetriever usage
+retriever = RaptorRetreiver(..., vector_store=vector_store)
```
Comment (jerryjliu) [inline llama-index-packs/llama-index-packs-raptor/README.md:67]: a few instances where *RaptorRetreiver --> *RaptorRetriever :)

**Review by jerryjliu — COMMENTED** [review]
sweet! some minor comments

**Review by nerdai — APPROVED** [review]
Generally, the code lgtm!

A couple of questions tho:

1. I thought maybe it would have made sense to implement the data processing step with clustering as a custom TransformComponent that can then be used in an `IngestionPipeline`. Any benefits in doing this? One that I think of is just to continue to enforce/promote usage of IngestionPipeline (in cases that fit well with it ofc, and I do think data processing / transforming is its primary raison d'etre?

2. Would it be possible to reproduce results of RAPTOR paper with their experiments using this pack implementation?

**Pull request merged**